java/src/main/java/io/grpc/examples/wallet/WalletServer.java

```diff
@@ -170,31 +171,19 @@ private void start() throws IOException {
         credentialsType == CredentialsType.XDS
             ? XdsServerCredentials.create(InsecureServerCredentials.create())
             : InsecureServerCredentials.create();
-    if (credentialsType == CredentialsType.XDS) {
-      server =
-          XdsServerBuilder.forPort(port, serverCredentials)
-              .addService(
-                  ServerInterceptors.intercept(
-                      new WalletImpl(accountChannel, statsChannel, v1Behavior),
-                      new WalletInterceptors.HostnameInterceptor(),
-                      new WalletInterceptors.AuthInterceptor()))
-              .addService(ProtoReflectionService.newInstance())
-              .addService(health.getHealthService())
-              .build()
-              .start();
-    } else {
-      server =
-          ServerBuilder.forPort(port)
-              .addService(
-                  ServerInterceptors.intercept(
-                      new WalletImpl(accountChannel, statsChannel, v1Behavior),
-                      new WalletInterceptors.HostnameInterceptor(),
-                      new WalletInterceptors.AuthInterceptor()))
-              .addService(ProtoReflectionService.newInstance())
-              .addService(health.getHealthService())
-              .build()
-              .start();
-    }
+
```

Comment:
Why did you delete this?

Reply:
Can use the same code path for the two `credentialsType`s. Previously `InsecureServerCredentials.create()` was not used.

Reply:
Are `XdsServerBuilder` and `ServerBuilder` the same?
Doesn't `XdsServerBuilder` wait for LDS? Or that behavior depends on the creds type?

Reply:
For 1.40 they are the same. From next release, it will wait for LDS. What is the plan? Do we want NOT to apply FilterChain if creds type is "insecure" even in the future releases?

cc @sanjaypujare

Reply:
I changed it back, not making unrelated change.